Application/src/view/visuals/component/Message.java

```diff
@@ -233,13 +239,35 @@ public void renderDefault(GraphicsContext gc) {
 
 
         //Draw trail for the message, for each instance in the arraylist
-        for (Trail t: trails)
-            gc.drawImage(trail, t.getXcoordinate(), (t.getYcoordinate() + 18), t.getWidth(), t.getHeight());
+        for (int i = 0; i < trails.size()-1; i++){
```

Comment:
How often does this happen?

Reply:
@timjon every time we draw a trail

Reply:
Yea, I double checked and thats true. The problem is that we are drawing the trail every fram. 

So each time we draw a trail you draw the arrow every time a fart in the trail gets drawn, and that is way too many times since it would be more than enough to draw it once for everytime the entire trail is drawn

Reply:
@timjon so you are not a fan of the arrow following the dragon? i think that is a cool feature :D 
otherwise changing it would need to change the logic entirely.
let's compensate here boss. i will make the arrow slightly bigger so it covers the fart behind it, then it won't look ugly just like u said.

Reply:
Wouldnt you get the same result by putting the stuff for drawing the arrow after the for loop?

Reply:
@timjon you forgot that i need each index/trail :) so it needs to be in the for loop

Reply:
@timjon but tbh i had the same idea when i was developing it haha

Reply:
Has this been tested before being dismissed?

Reply:
@Sebastian-Fransson yes sir